overlord/snapstate/storehelpers.go

```diff
@@ -613,7 +613,88 @@ func collectCurrentSnaps(snapStates map[string]*SnapState, holds map[string][]st
 	return curSnaps, nil
 }
 
+// refreshCandidates is a wrapper for refreshCandidatesCore.
+//
+// It addresses the case were the store doesn't return refresh candidates for
+// snaps with already existing refresh-candidates that are being monitored due
+// to throttling.
+// A second request is sent for eligible snaps that might have been throttled
+// with the RevisionOptions.Scheduled option turned off.
+//
+// Note: This wrapper is a short term solution and should be removed once a better
+// solution is reached.
 func refreshCandidates(ctx context.Context, st *state.State, names []string, revOpts []*RevisionOptions, user *auth.UserState, opts *store.RefreshOptions) ([]*snap.Info, map[string]*SnapState, map[string]bool, error) {
+	updates, stateByInstanceName, ignoreValidation, err := refreshCandidatesCore(ctx, st, names, revOpts, user, opts)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var oldHints map[string]*refreshCandidate
+	if err := st.Get("refresh-candidates", &oldHints); err != nil {
+		if errors.Is(err, &state.NoStateError{}) {
+			// do nothing
+			return updates, stateByInstanceName, ignoreValidation, nil
+		}
+
+		return nil, nil, nil, fmt.Errorf("cannot get refresh-candidates: %v", err)
+	}
+
+	var missingNames []string
+	snapStates, err := All(st)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	for name := range oldHints {
+		if _, ok := snapStates[name]; !ok {
+			continue
+		}
+		if !snapStates[name].IsInstalled() {
+			continue
+		}
+		if !isSnapMonitored(st, name) {
+			continue
+		}
+		hasUpdate := false
+		for _, update := range updates {
+			if update.InstanceName() == name {
+				hasUpdate = true
+				break
+			}
+		}
+		if hasUpdate {
+			continue
+		}
+
+		missingNames = append(missingNames, name)
+	}
+
+	if len(missingNames) > 0 {
+		var revOptsByName map[string]*RevisionOptions
+		var missingRevOpts []*RevisionOptions
+		if revOpts != nil {
+			revOptsByName = make(map[string]*RevisionOptions, len(revOpts))
+			for i, opts := range revOpts {
+				revOptsByName[names[i]] = opts
+			}
+			for _, name := range missingNames {
+				missingRevOpts = append(missingRevOpts, revOptsByName[name])
+			}
```

Comment:
It'd be good to have a test with `revOpts != nil` to exercise this code

Reply:
Great catch, I updated to reuse the test from [this PR](https://github.com/snapcore/snapd/pull/13265/files). Turns out there was a bug due `refreshCandidatesCore` mutating `names` while sorting. Thankss!